index.html

```diff
@@ -228,6 +228,66 @@ <h4>New sink available</h4>
+    <h2><a><code>MediaDevices</code></a> Extensions</h2>
+    <p>This section specifies additions to the <dfn
+    data-cite="!GETUSERMEDIA#dom-mediadevices"><code>
+    MediaDevices</code></dfn> when the Audio Output Devices API is
+    supported.</p>
+    <div>
+      <pre class="idl">
+      partial interface MediaDevices {
+        Promise&lt;MediaDeviceInfo&gt; selectAudioOutput();
+      };
+      </pre>
+      <section>
+        <h2>Methods</h2>
+        <!-- FIXME: Update section to use links to promise resolve/reject spec, permission spec... -->
+        <dl data-link-for="MediaDevices" data-dfn-for="MediaDevices"
+        class="method">
+          <dt><dfn><code>selectAudioOutput</code></dfn></dt>
+          <dd>
+            <p>Prompts the user to select a specific audio output device.</p>
+            <p>When the <code>selectAudioOutput</code> method is called,
+            the User Agent MUST run the following steps:</p>
+            <ol>
+              <li><p>If the method call is not <a
+              href="https://html.spec.whatwg.org/multipage/interaction.html#triggered-by-user-activation">
+              triggered by user activation</a>,
+              return a promise rejected with a <code>DOMException</code> object whose
+              <code>name</code> attribute has the value <code>InvalidStateError</code>.
+              </p></li>
+              <li><p>Let <var>p</var> be a new promise.</p></li>
+              <li>
+                <p>Run the following steps in parallel:</p>
+                <ol>
+                  <li><p>Probe the User Agent for available audio output devices.</p></li>
+                  <li><p>If there is no audio output device, reject
+                  <var>p</var> with a new <code>DOMException</code> whose <code>name</code> attribute
+                  has the value <code>NotFoundError</code>.</p></li>
+                  <li><p>Prompt the user to choose an audio output device, with a PermissionDescriptor named "speaker".</p></li>
+                  <li><p>If the result of the request is "denied", reject
+                    <var>p</var> with a new <code>DOMException</code> whose <code>name</code> attribute
+                    has the value <code>NotAllowedError</code> and abort these steps.</p></li>
+                  <li><p>Let <var>deviceInfo</var> be a new <code>MediaDeviceInfo</code> object to represent the selected audio output device.</p></li>
+                  <li><p>Resolve <var>p</var> with <var>deviceInfo</var>.</p></li>
+                </ol>
+              </li>
+              <li><p>Return <var>p</var>.</p></li>
+            </ol>
+            <!-- FIXME: Introduce necessary hooks to expose a device in mediacapture main spec
+            and use it as a specific step in above algorithm. -->
+            <p>Once a device is exposed after a call to <code>selectAudioOutput</code>, it MUST be listed by
+            <code>enumeratedevices</code> for the current browsing context.</p>
```

Comment:
enumerateDevices

Probably a hook needed so that enumerateDevices algorithm references a list modified by this algorithm.